src/ray/core_worker/transport/raylet_transport.cc

```diff
@@ -26,6 +28,12 @@ void CoreWorkerRayletTaskReceiver::HandleAssignTask(
   const raylet::Task task(request.task());
   const auto &spec = task.GetTaskSpecification();
   auto status = task_handler_(spec);
+  // Notify raylet the current task is done. This is to ensure that the task
+  // is marked as finished by raylet only after previous raylet client calls are
+  // completed. The rpc `done_callback` is sent via a different connection
+  // from raylet client connection, so it cannot guarantee the rpc reply arrives
+  // at raylet after a previous `NotifyUnblocked` message.
```

Comment:
I'm a little confused by this comment...the second sentence makes it sound like the raylet will not get this callback until after all other calls have completed, but then the third sentence seems to say the opposite? What exactly is the guarantee of `TaskDone`?

Reply:
Sorry for the confusion. The third sentence is actually saying this is not possible with rpc `done_callback`, and that's why we need to use `TaskDone`. I'll update the comments and make it clear.